systemtests/src/test/java/io/enmasse/systemtest/bases/clients/ClientTestBase.java

```diff
@@ -170,7 +171,19 @@ protected void doRoundRobinReceiverTest(AbstractClient sender, AbstractClient re
         Future<Boolean> recResult = receiverClient1.runAsync();
         Future<Boolean> rec2Result = receiverClient2.runAsync();
 
-        Thread.sleep(15000);
+        Thread.sleep(2);
+
+        if (receiverClient1.getClientAttachedProbe() != null) {
```

Comment:
Couldn't you just make `getClientAttachedProbe() ` return a default implementation whose get() call simply sleeps for the duration of the timeout argument.  That'd get rid of the conditions.

Reply:
I like the idea, done